packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -102,17 +102,17 @@ class CurlyComponentManager {
 
     assert('You cannot use `classNameBindings` on a tag-less component: ' + component.toString(), () => {
       let { classNameBindings, tagName } = component;
-      return tagName || !classNameBindings || classNameBindings.length === 0;
+      return tagName !== '' || !classNameBindings || classNameBindings.length === 0;
     });
 
     assert('You cannot use `elementId` on a tag-less component: ' + component.toString(), () => {
       let { elementId, tagName } = component;
-      return tagName || (!elementId && elementId !== '');
+      return tagName !== '' || (!elementId && elementId !== '');
```

Comment:
Fixed in https://github.com/emberjs/ember.js/pull/13684/commits/e0672cde9aa33e0134075f5a1fb26b06a0b40ef2